lib/client.js

```diff
@@ -308,6 +319,12 @@ MqttClient.prototype._setupStream = function () {
 }
 
 MqttClient.prototype._handlePacket = function (packet, done) {
+  var options = this.options
+
+  if (options.protocolVersion === 5 && options.properties && options.properties.maximumPacketSize && options.properties.maximumPacketSize < packet.length) {
+    this.emit('error', new Error('exceeding packets size ' + packet.cmd))
```

Comment:
if we are emitting error, we should really destroy the client or fully disconnect (and reconnect).

Reply:
i`ll add end for connection with reason code, but don`t add reconnect, because I get loop, if do this.

Reply:
From spec: If a Client receives a packet whose size exceeds this limit, this is a Protocol Error, the Client uses DISCONNECT with Reason Code 0x95 (Packet too large). If you think, that bad idea, I can remove emitting error, but I think thats right. Just tell me what do you think. I think user must proccess that error himself.